parser/src/ast/mod.rs

```diff
@@ -60,8 +64,74 @@ pub struct TraceCols {
 #[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Clone)]
 pub struct Identifier(pub String);
 
+impl Identifier {
+    /// Returns the name of the identifier.
+    pub fn name(&self) -> &str {
+        &self.0
+    }
+}
+
 impl fmt::Display for Identifier {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{}", &self.0)
     }
 }
+
+/// [VectorAccess] is used to represent an element inside vector at the specified index.
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Clone)]
+pub struct VectorAccess {
+    name: Identifier,
+    idx: usize,
+}
+
+impl VectorAccess {
+    /// Creates a new [VectorAccess] instance with the specified identifier name and index.
+    pub fn new(name: Identifier, idx: usize) -> Self {
+        Self { name, idx }
+    }
+
+    /// Returns the name of the vector.
+    pub fn name(&self) -> &str {
+        self.name.name()
+    }
+
+    /// Returns the index of the vector access.
+    pub fn idx(&self) -> usize {
+        self.idx
+    }
+}
+
+/// [MatrixAccess] is used to represent an element inside a matrix at the specified row and column
+/// indices.
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Clone)]
+pub struct MatrixAccess {
+    name: Identifier,
+    row_idx: usize,
+    col_idx: usize,
+}
+
+impl MatrixAccess {
+    /// Creates a new [MatrixAccess] instance with the specified identifier name and indices.
+    pub fn new(name: Identifier, col_idx: usize, row_idx: usize) -> Self {
+        Self {
+            name,
+            row_idx,
+            col_idx,
+        }
+    }
+
+    /// Returns the name of the matrix.
+    pub fn name(&self) -> &str {
+        &self.name.0
```

Comment:
This should also be `&self.name.name()` now that we have it

Reply:
Ah sorry, I thought I changed both instances. Fixed.